api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -252,5 +257,102 @@ else if (!isObsGroup) {
 			}
 		}
 	}
+
+	/**
+	 * This method validates Obs' concept values.
+	 * 
+	 * <ol>
+	 *     <li>Validates if high absolute and low absolute are within the valid range</li>
+	 *     <li>Validates if patient's age is within the valid range</li>
+	 * <ol/>
+	 * in the {@link ConceptReferenceRange}.
+	 *
+	 * @param obs Observation to validate
+	 * @param errors Errors to record validation issues
+	 *               
+	 * @since 2.7.0
+	 */
+	private void validateConceptReferenceRange(Obs obs, Errors errors) {
+		Concept concept = obs.getConcept();
+
+		if (concept != null && concept.getDatatype() != null && concept.getDatatype().isNumeric()) {
+			List<ConceptReferenceRange> referenceRanges = Context.getConceptService()
+				.getConceptReferenceRangesByConceptId(concept.getConceptId());
+
+			if (!referenceRanges.isEmpty()) {
+				ConceptReferenceRangeUtility utility = new ConceptReferenceRangeUtility();
+				List<ConceptReferenceRange> validRanges = new ArrayList<>();
+
+				for (ConceptReferenceRange referenceRange : referenceRanges) {
+					if (utility.evaluateCriteria(referenceRange.getCriteria(), obs.getPerson())) {
+						validRanges.add(referenceRange);
+					}
+				}
+
+				if (!validRanges.isEmpty()) {
+					ConceptReferenceRange strictestRange = validRanges.stream()
```

Comment:
Do we have tests for this?

Reply:
Strictness is considered not for the whole range, but rather each individual lower or upper bound.
For instance if we have these two ranges for criteria that match: `20 - 50` and `10 - 30`, for the lower bound, `20` is more strict. And for the upper bound, `30` is more strict. Therefore, this would result into the obs value being validated against a range of `20 - 30`. Does this make sense? 😊